packages/core/src/contracts/ownedAddressTCRWithAppeals.ts

```diff
@@ -0,0 +1,398 @@
+  public static atUntrusted(
+    web3wrapper: Web3Wrapper,
+    contentProvider: ContentProvider,
+    address: EthAddress,
+  ): OwnedAddressTCRWithAppeals {
+    const instance = OwnedAddressTCRWithAppealsContract.atUntrusted(web3wrapper, address);
+    return new OwnedAddressTCRWithAppeals(web3wrapper, contentProvider, instance);
+  }
+
+  private contentProvider: ContentProvider;
+
+  private constructor(
+    web3Wrapper: Web3Wrapper,
+    contentProvider: ContentProvider,
+    instance: OwnedAddressTCRWithAppealsContract,
+  ) {
+    super(web3Wrapper, instance);
+    this.contentProvider = contentProvider;
+  }
+
+  /**
+   * Event Streams
+   */
+
+  /**
+   * An unending stream of all addresses currently whitelisted
+   * @param fromBlock Starting block in history for events concerning whitelisted addresses.
+   *                  Set to "latest" for only new events
+   * @returns currently Whitelisted addresses
+   */
+  public whitelistedListings(fromBlock: number|"latest" = 0): Observable<EthAddress> {
+    return this.instance
+      .NewListingWhitelistedStream({}, { fromBlock })
+      .distinctUntilChanged((a, b) => {
+        return a.blockNumber === b.blockNumber && a.logIndex === b.logIndex;
+      })
+      .map((e) => e.args.listingAddress)
+      .concatFilter(async (listingAddress) => this.instance.getListingIsWhitelisted.callAsync(listingAddress));
+  }
+
+  /**
+   * An unending stream of all addresses currently applied
+   * @param fromBlock Starting block in history for events concerning applied addresses.
+   *                  Set to "latest" for only new events
+   * @returns currently applied addresses
+   */
+  public currentAppliedListings(fromBlock: number|"latest" = 0): Observable<EthAddress> {
+    return this.instance
+      .ApplicationStream({}, { fromBlock })
+      .distinctUntilChanged((a, b) => {
+        return a.blockNumber === b.blockNumber && a.logIndex === b.logIndex;
+      })
+      .map((e) => e.args.listingAddress)
+      .concatFilter(async (listingAddress) => this.isInUnchallengedApplicationPhase(listingAddress));
+  }
+
+  /**
+   * An unending stream of all addresses currently able to be whitelisted
+   * @param fromBlock Starting block in history for events concerning ready to whitelist addresses.
+   *                  Set to "latest" for only new events
+   * @returns addresses ready to be whitelisted
+   */
+  public readyToBeWhitelistedListings(fromBlock: number|"latest" = 0): Observable<EthAddress> {
+    return this.instance
+    .ApplicationStream({}, { fromBlock })
+    .distinctUntilChanged((a, b) => {
+      return a.blockNumber === b.blockNumber && a.logIndex === b.logIndex;
+    })
+    .map((e) => e.args.listingAddress)
+    .concatFilter(async (listingAddress) => this.instance.canBeWhitelisted.callAsync(listingAddress));
+  }
+
+  /**
+   * An unending stream of all addresses currently challenged in commit vote phase
```

Comment:
Please explain to our users briefly what are the phases and how they work in the class comment